internal/interface/handlers/document_handler.go

```diff
@@ -103,7 +103,56 @@ func (h *DocumentHandler) GetDocumentByID(c *fiber.Ctx) error {
 // @Failure 500 {object} response.Response
 // @Router /documents/role/{role_id} [get]
 func (h *DocumentHandler) GetDocumentsByRole(c *fiber.Ctx) error {
-	return nil
+	// validate parameter
+	getallDocumentsByRoleDTO := dtos.GetAllDocumentsByRoleDTO{
+		Page:         c.QueryInt("page", 1),
+		PageSize:     c.QueryInt("page_size", 10),
+		Title:        c.Query("title"),
+		DocumentType: c.Query("document_type"),
+		Organization: c.Query("organization"),
+		Role:         c.Params("role_id"),
```

Comment:
The role field is smt like `SCCU_ADMIN` right? Let's change it to `c.Params("role")` for more consistency. You can update the swagger by using `swag init -d ./internal/interface/handlers -g ../../../cmd/main.go -o ./docs -md ./docs/markdown --parseDependency --parseInternal` in root dir

Reply:
updated!!